src/trace.c

```diff
@@ -1051,13 +1080,13 @@ trace_add_path (const void *parent,
 		nih_hash_add (path_hash, &entry->entry);
 	}
 
-	/* Make sure that we have an ordinary file
-	 * This avoids us opening a fifo or socket or symlink.
+	/* Make sure again that we have an ordinary file
```

Comment:
Can I ask why we should check it again?

Reply:
Initial `lstat` returns an information of symlink instead of the file that link points to.
If the code goes through `if (S_ISLNK(...))` block, we still don't know if the file which symlink points to is a regular file or not.

Let me move this check inside `if (S_ISLNK(...))` block since this check is only relevant when the code goes through this path.

Reply:
Done.